src/components/dialog/dialog.js

```diff
@@ -646,9 +646,16 @@ function MdDialogProvider($$interimElementProvider) {
       }
       if (options.clickOutsideToClose) {
         var target = element;
-        var clickHandler = function(ev) {
-          // Only close if we click the flex container outside on the backdrop
-          if (ev.target === target[0]) {
+        var sourceElem;
+
+        var mousedownHandler = function(ev) {
+          // Saving the source element for later checking if it was the backdrop
```

Comment:
This is more along the lines of the explanation I'm looking for:
```
Keep track of the element on which the mouse originally went down so that we can only close the backdrop when the "click" started on it. A simple 'click' handler does not work because...
```

(filling in why `click` doesn't work)